repository/scripts/widget/frame/minimap.js

```diff
@@ -44,98 +44,104 @@ module.exports = (function () {
-				case 4://Withdraw money pouch
-					moneyPouch.requestWithdrawCoins(player);
-					return;
-				case 6://Bond pouch
-					return;//This is client-sided
-				//case 5://Wealth evaluator
-					//player.getWidgets().openWidget(1477, 380, 566, false);
-				}	
-				util.defaultHandler(ctx, "minimap");
-				return;		
-			case 44://World map
-				varp(player, 3926, 0);
-				varp(player, 3928, -1);
-				varp(player, 3929, -1);
-				widget.open(player, 1477, 496, 669, true);
-				varc(player, 3838, 0);
-				varc(player, 3840, 0);
-				widget.hide(player, 1477, 20, true);
-				widget.hide(player, 1477, 21, true);
-				widget.hide(player, 1477, 22, true);
-				widget.hide(player, 1477, 506, true);
-				widget.hide(player, 1477, 388, true);
-				//TODO: Find out the right varp for these
-				//varp(player, 622, api.getCoordHash(api.getCoords(player)));
-				//varp(player, 674, api.getCoordHash(api.getCoords(player)));
-				widget.open(player, 1477, 16, 1421, true);
-				widget.open(player, 1477, 15, 1422, false);
-				widget.open(player, 1422, 107, 698, true);
-				/*player.getDispatcher().sendWidgetSettings(1422, 38, 2, 2, 2);
-				player.getDispatcher().sendWidgetSettings(1422, 39, 2, 2, 2);
-				player.getDispatcher().sendWidgetSettings(1422, 40, 2, 2, 2);
-				player.getDispatcher().sendWidgetSettings(1422, 41, 2, 2, 2);
-				player.getDispatcher().sendWidgetSettings(1422, 42, 2, 2, 2);
-				player.getDispatcher().sendWidgetSettings(1422, 43, 2, 2, 2);
-				player.getDispatcher().sendWidgetSettings(1422, 44, 2, 2, 2);
-				player.getDispatcher().sendWidgetSettings(1422, 45, 2, 2, 2);
-				player.getDispatcher().sendWidgetSettings(1422, 46, 2, 2, 2);
-				player.getDispatcher().sendWidgetSettings(1422, 47, 2, 2, 2);
-				player.getVars().setVarp(622, player.getCurrentTile().getTileHash());
-				player.getDispatcher().sendWidgetSettings(1422, 86, 0, 19, 2);
-				player.getDispatcher().sendHideWidget(1422, 49, true);
-				player.getDispatcher().sendVarc(4197, -1);
-				player.getVars().setVarp(674, player.getCurrentTile().getTileHash());*/
-				return;		
-			case 46://Toggle run
-				if (ctx.button == 1) {
-					var running = varp(player, 463) == 1;
-					if (running) {
-						varp(player, 463, 0);
-						//TODO: Create an engine method for this
-						player.getMovement().setRunning(false);
-					} else {
-						varp(player, 463, 1);
-						player.getMovement().setRunning(true);
-					}
-				} else if (ctx.button == 2) {
-					anim.run(player, 5713);
-					chat.sendDebugMessage(player, "Unhandled rest action...");
-				}
-				return;
-			case 57://Open lodestone interface
-				widget.openCentral(player, 1092, false);
-				return;
-			/*case 14://Legacy XP counter button
-			case 27://Open notes
-			case 30://Open group system
-			case 33://Open metrics
-			case 55://Reset camera*/
-			default:
-				util.defaultHandler(ctx, "minimap");
-				return;
-			}
-		});
+	scriptManager.bind(EventType.IF_BUTTON, 1465, function (ctx) {
+	var player = ctx.player;
```

Comment:
I just added 
case 44://World map
	if (ctx.button == 3) {
	varp(player, 2250, 1073741824);
	varc(player, 2771, 53038235);
	widget.openCentral(player, 1587, false);
	} else {
for world select